lib/block/block.go

```diff
@@ -71,19 +73,19 @@ func NewBlock(proposer string, round round.Round, transactions []string, confirm
 		Confirmed:    confirmed,
 	}
 
-	log.Debug("NewBlock created", "PrevTotalTxs", round.TotalTxs, "txs", len(transactions), "TotalTxs", b.Header.TotalTxs)
-
 	b.Hash = base58.Encode(common.MustMakeObjectHash(b))
+	log.Info("NewBlock created", "block", b)
```

Comment:
:+1:

Reply:
While the change in the test is nice, I feel like a constructor should not log stuff. It should be at the point where things are externalized.

Reply:
@Geod24 I want to know the reason.

Reply:
- https://dave.cheney.net/2017/01/23/the-package-level-logger-anti-pattern

Reply:
A constructor should strive to be as [pure](https://en.wikipedia.org/wiki/Pure_function) as possible. 

In this case, you can call the constructor many times, e.g. in tests. Do you really want to log every time you call this constructor from a test ?
Likewise, we probably only invoke this ctor once at the moment. If that change, this error message will become very noisy, won't it ?

Reply:
IMHO, The `pure` term belongs to Functional language. As you know, Dlang has `pure` keyword like this `pure int foo (...)` But, I'm not sure that gophers consider `pure function` when they do. :-> 
Otherwise, I would like to `NilLogger` or `TestLogger` when I test. 

- https://github.com/tendermint/tendermint/blob/master/blockchain/pool_test.go#L33-40
- https://github.com/tendermint/tendermint/blob/46369a1ab76f274ab47179c4176221842b8207b4/docs/architecture/adr-001-logging.md